contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,573 @@
+            )
+            : dexSwaps(getSwapAddress()).dexAmountOut(
+                order.base,
+                order.loanTokenAddress,
+                order.collateralTokenAmount
+            );
+        if (fSwapRate == 0) {
+            return 0;
+        }
+        return
+            order.orderType == IWalletFactory.OrderType.LIMIT_OPEN
+                ? (tradeSize.TenExp(18 - int8(IERC20Metadata(order.loanTokenAddress).decimals())) * 1 ether) /
+                    (fSwapRate.TenExp(18 - int8(IERC20Metadata(order.base).decimals())))
+                : (1 ether *(fSwapRate.TenExp(18 - int8(IERC20Metadata(order.loanTokenAddress).decimals())))) /
+                    (order.collateralTokenAmount.TenExp(18 - int8(IERC20Metadata(order.base).decimals())));
+    }
+
+    function dexSwapCheck(
+        uint256 collateralTokenAmount,
+        uint256 loanTokenAmount,
+        address loanTokenAddress,
+        address base,
+        uint256 leverage,
+        IWalletFactory.OrderType orderType
+    ) public view returns (uint256) {
+        uint256 tradeSize;
+        if (orderType == IWalletFactory.OrderType.LIMIT_OPEN) {
+            if (loanTokenAmount > 0) {
+                tradeSize = (loanTokenAmount * leverage) / 1 ether;
+            } else {
+                (tradeSize, ) = dexSwaps(getSwapAddress()).dexAmountOut(
+                    base,
+                    loanTokenAddress,
+                    collateralTokenAmount
+                );
+                if (tradeSize == 0) {
+                    return 0;
+                }
+                tradeSize = (tradeSize * leverage) / 1 ether;
+            }
+        }
+        (uint256 fSwapRate, ) = orderType == IWalletFactory.OrderType.LIMIT_OPEN
+            ? dexSwaps(getSwapAddress()).dexAmountOut(
+                loanTokenAddress,
+                base,
+                tradeSize
+            )
+            : dexSwaps(getSwapAddress()).dexAmountOut(
+                base,
+                loanTokenAddress,
+                collateralTokenAmount
+            );
+        if (fSwapRate == 0) {
+            return 0;
+        }
+        return
+            orderType == IWalletFactory.OrderType.LIMIT_OPEN
+                ? (tradeSize.TenExp(18 - int8(IERC20Metadata(loanTokenAddress).decimals())) *
+                    1 ether) /
+                    (fSwapRate.TenExp(18 - int8(IERC20Metadata(base).decimals())))
+                : (1 ether *
+                    (fSwapRate.TenExp(18 -
+                                int8(IERC20Metadata(loanTokenAddress).decimals())))) /
+                    (collateralTokenAmount.TenExp(18 - int8(IERC20Metadata(base).decimals())));
+    }
+
+    function gasPrice(address payToken) public view returns (uint256) {
+        return IPriceFeeds(getFeed()).getFastGasPrice(payToken) * 2;
+    }
+
+	function clearOrder(address trader, uint256 orderID)
+		public
+		view
+		returns (bool)
+	{
+		if(orderExpiration[trader][orderID] < block.timestamp){
```

Comment:
usually there is a special type of order called GTD which stands good till date. I am just saying orderExpiration might be part of order Struct, we can have GTC or maybe others in future